monai/networks/nets/segresnet_ds.py

```diff
@@ -0,0 +1,476 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from monai.networks.blocks.upsample import UpSample
+from monai.networks.layers.factories import Act, Conv, Norm, split_args
+from monai.utils import UpsampleMode, has_option
+
+__all__ = ["SegResNetDS"]
+
+
+def get_norm_layer(name: Union[Tuple, str], spatial_dims: Optional[int] = 1, channels: Optional[int] = 1):
```

Comment:
Could you please reuse the existing api? or help improve the existing ones

Reply:
I have tried to update existing API, e.g. for get_norm_layer() here is my PR https://github.com/Project-MONAI/MONAI/pull/5220

But it seems to have no support, since it'll trigger changes for all legacy models.  looks liek it's easier to add self-contained methods here, to prevent issues with legacy dependencies.

Reply:
> looks liek it's easier to add self-contained methods here, to prevent issues with legacy dependencies.

it's easier for this PR but adds maintenance burdens if we all follow this pattern of adding different variants of the same thing...

Reply:
Sure, but what's the solution. Would you like to approve https://github.com/Project-MONAI/MONAI/pull/5220, then I can reuse existing functions, but it'll break previous models.

Reply:
Any other comments/suggestions? If I do change get_norm_layer() and get_act_layer() for the main API it will break the functionality for legacy networks (not just segresnet).   https://github.com/Project-MONAI/MONAI/pull/5220

It seems it will simplify your work not to retest everything due to this small change. I prefer the model to be self-contained.

Reply:
Following your suggestions, I've changed the code to use existing  get_norm_layer, get_act_layer, and set the defaults for my code inside of the network itself.